README.md

```diff
@@ -380,7 +402,7 @@ method takes two arguments:
 ### Profunctor
 
 A value that implements the Profunctor specification must also implement
-the Functor specification.
+the Functor and Contravariant Functor specifications.
```

Comment:
> Am I write in thinking that there could be a class AmateurFunctor with the opposite problem?

In something like Haskell, a type that implements `AmateurFunctor`  could only be declared as `Contravariant`.
```hs
class AmateurFunctor f where
  amateurmap :: (a -> b) -> (c -> d) -> f a d -> f b c

class Contravariant f where
  contramap :: (b -> a) -> f a -> f b

instance AmateurFunctor f => Contravariant (f a) where
  contramap = amateurmap id
```

To also declare it as a `Functor` you would need to flip the order of the type arguments with a `newtype`.
```hs
newtype Flip f b a = Flip { unFlip :: f a b }

instance AmateurFunctor f => Functor (Flip f a) where
  fmap f = Flip . amateurmap f id . unFlip
```
If however we were to ignore the notion of type arguments, it could just be stated that:
```js
p.promap(f, g) = p.contramap(f).map(g) = p.map(g).contramap(f)
```
It looks a little odd for me to see `contramap` and `map` being applied directly to the same type, but as the `Flip` newtype above highlights, the types are effectively identical.